src/server/routes/compareReadings.js

```diff
@@ -10,6 +10,16 @@ const moment = require('moment');
 const { getConnection } = require('../db');
 
 const Reading = require('../models/Reading');
+const { GENERAL_STRING_MAX_LENGTH, NUMERIC_ID_MAX_LENGTH } = require('../util/validationConstants');
+const ISO_DURATION_REGEX = /^P(?!$)(\d+Y)?(\d+M)?(\d+D)?(T(\d+H)?(\d+M)?(\d+(\.\d+)?S)?)?$/;
+
+function isValidIsoDateTime(value) {
+	return typeof value === 'string' && moment.parseZone(value, moment.ISO_8601, true).isValid();
```

Comment:
I see the value of checking if it is a string. However, doesn't the param validation do this before this runs?

Also, can you help me understand why this particular moment function/params are used to check this?